docs/guides/migrations/guide.rst

```diff
@@ -0,0 +1,1298 @@
+about EdgeDB Watch, which is ever more effortless!) This leads to 
+an interesting side effect: lots of ``.edgeql`` files, many of which 
+represent trials and approaches that don't end up making it to the 
+final form for your schema.
+
+Once you are done, you might want to squash the migrations into a 
+single file. This is especially nice if you need to frequently initialize 
+database instances using the same schema, because all migrations are 
+applied when an instance starts up. You can imagine that the output 
+would be pretty long if you had dozens and dozens of migration files 
+to work through:
+
+.. code-block::
+
+    Initializing EdgeDB instance...
+    Applying migrations...
+    Applied m13brvdizqpva6icpcvmsc3fee2yt5j267uba6jugy6iugcbs2djkq 
+    (00001.edgeql)
+    Applied m1aildofb3gvhv3jaa5vjlre4pe26locxevqok4semmlgqwu3xayaa 
+    (00002.edgeql)
+    Applied m1ixxlsdgrlinfijnrbmxdicmpfav33snidudqi7fu4yfhg4nngoza 
+    (00003.edgeql)
+    Applied m1tsi4amrdbcfjypu72duyckrlvvyb46r3wybd7qnbmem4rjvnbcla 
+    (00004.edgeql)
+    ...and so on...
+    Project initialized.
+
+To squash your migrations, just add ``--squash`` after ``edgedb migration 
+create``. Running this command will first display some helpful info 
+to keep in mind before committing to the operation:
+
+.. code-block::
+
+    Current database revision is: 
+    m16ixoukn7ulqdn7tp6lvx2754hviopanufv2lm6wf4x2borgc3g6a
+    While squashing migrations is non-destructive, 
+    it may lead to manual work if done incorrectly.
+
+    Items to check before using --squash:
+    1. Ensure that `./dbschema` dir is comitted
+    2. Ensure that other users of the database have the revision
+    above or can create database from scratch.
+        To check a specific instance, run:
+        edgedb -I <name> migration log --from-db --newest-first --limit 1
+    1. Merge version control branches that contain schema changes
+    if possible.
+
+    Proceed? [y/n]
+
+Then just press ``y`` to squash all of your existing migrations into 
+a single file.
+
+Fixups during a squash
+----------------------
+
+If your schema doesn't match the schema in the database, EdgeDB will 
+prompt you to create a *fixup* file, which can be useful (as the CLI 
+says) to "automate upgrading other instances to a squashed revision". 
+You'll see fixups inside a folder called ``/fixups``. Their file names 
+are extremely long, but are simply two migration hashes joined together 
+by a dash. So a fixup that begins with
+
+.. code-block::
+
+    CREATE MIGRATION 
+    m1v3vqmwif4ml3ucbzi555mjgm4myxs2husqemopo2sz2m7otr22ka 
+    ONTO m16awk2tzhtbupjrzoc4fikgw5okxpfnaazupb6rxudxwin2qfgy5q
+
+will have a file name a full 116 characters in length.
+
+The CLI output when using squash along with a fixup is pretty informative 
+on its own, so let's just walk through the output as you'll see it 
+in practice.
+
+First we'll begin with this schema:
```

Comment:
```suggestion
in practice. First we'll begin with this schema:
```

Reply:
Also, we should specify how we want the reader to follow along here. Should they start a new project or migrate the project we created earlier to this new schema?